File: app/src/main/java/com/yalantis/contract/BaseMvpView.java
package com.yalantis.contract;

import android.content.Context;
import android.support.annotation.StringRes;

public interface BaseMvpView {

    Context getContext();

    void showProgress();

    void hideProgress();

    void showError(@StringRes int strResId);

    void showError(String error);

    void showMessage(@StringRes int srtResId);

    void showMessage(String message);

}

File: app/src/main/java/com/yalantis/base/BaseMvpView.kt
package com.yalantis.base

import android.content.Context
import android.support.annotation.StringRes
import com.trello.navi.NaviComponent

interface BaseMvpView : NaviComponent {

    fun getContext(): Context

    fun showProgress()

    fun hideProgress()

    fun showError(@StringRes strResId: Int)

    fun showError(error: String)

    fun showMessage(@StringRes srtResId: Int)

    fun showMessage(message: String)

}
